## Inference

### Packages

In [14]:
import torch
import numpy as np
import pandas as pd
import joblib

from sklearn.preprocessing import StandardScaler

### Importing the Model

In [13]:
model = torch.load("model_complete.pth")
model.eval()

Sequential(
  (0): Linear(in_features=7, out_features=24, bias=True)
  (1): ReLU()
  (2): Linear(in_features=24, out_features=16, bias=True)
  (3): ReLU()
  (4): Linear(in_features=16, out_features=6, bias=True)
  (5): ReLU()
  (6): Linear(in_features=6, out_features=1, bias=True)
)

### Race Prediction

#### 10k Turkey Trot
- Max Grade: 0.0013
- Average Cadence: 170
- Distance (mi): 6.2
- Max Speed (mph): 12
- Elevation Gain (ft): 43
- Elevation High (ft): 102
- Rest Hours: 36

*Using the saved scaler object that was fit on the training data*

In [21]:
scaler = joblib.load('scaler.pkl')

inputs = np.array([[0.0013, 170, 6.2, 12, 43, 102, 36]])
inputs = scaler.transform(inputs)
inputs = torch.tensor(inputs, dtype=torch.float32)
inputs

tensor([[-1.3484, 40.1209,  0.2967,  0.1542, -1.0593, -0.8138, -0.4192]])

### Prediction

In [27]:
model.eval()

with torch.no_grad():
    prediction = model(inputs)
    
print(prediction.numpy()[0][0], "Minutes")

44.52577 Minutes


I'd expect that, after following my training plan, my race pace will be about *15% faster* than my training run pace. 

#### 44.52577 minutes * 0.85 = 37.76190 minutes

#### 37.76190 minutes / 6.2 miles = 6.09063 minutes/mile

## Final Prediction: 6.09 minutes/mile
    
Let's check back in November and see how I do!